### PR TITLE
Automate release process

### PR DESCRIPTION
**Pull request by ekcorso**
This automates the process for releasing a new version of tldextract. It prompts the user to add a tag, creates a build, prompts for verification, uploads the build to PyPi, pushes the new tag to Github, and then creates a draft of the Github release. 

Note that the release notes will be based based on a combination of the CHANGELOG.md entry that matches the version tag and GitHub's generated changelog URL.

The user will need a Github app auth token and Pypi credentials in order to run this.

# Changes

* Add script to interactively release the current work tree, tagged, to PyPI and GitHub
* Add release-only dependencies
* Typecheck scripts folder
    * The folder is not a package, so tell mypy `--explicit-package-bases`
* Test script happy path
    * Add snapshot library dependency
        * To unit test network and subprocess input, and verbose CLI output
    * Skip test on Windows
        * Snapshot paths are platform-dependent. Maintainers don't yet release using Windows.

### CODE REVIEW

**Comment by john-kurkowski** [comment]
For PyPI releases of this project, I told @ekcorso I was trying to remember and manually execute several steps. We collaborated on this script offline. Now my manual steps are codified in this repo! Hopefully less error prone for me, and the steps could be followed by any maintainer. I'm excited to put this to use. Thank you!

File: scripts/release.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+- Create a build.
+- Ask the user to verify the build.
+- Upload the build to PyPI.
+- Push all git tags to the remote.
+- Create a draft release on GitHub using the version notes in CHANGELOG.md.
+
+Prerequisites:
+    - This must be run from the root of the repository.
+    - The repo must have a clean git working tree.
+    - The user must have the GITHUB_TOKEN environment variable set to a valid GitHub personal access token.
```
Comment (john-kurkowski) [inline scripts/release.py:16]: Can you add a bullet for PyPI credentials? During the script, users will need to paste in a token from their password manager?

**Review by john-kurkowski — APPROVED** [review]

File: scripts/release.py (hunk 1 above)
Comment (ekcorso) [reply scripts/release.py:16]: No problem. Done!

**Comment by john-kurkowski** [comment]
I used the script to release 5.1.2. Script worked like a charm! 🙌

Only a couple minor follow-ups:

1. I fat-fingered something later in the process and couldn't re-run the script as is, I had to manually delete a tag, outside the script. Automated tag rollback in  befc91c9c3a15dff464d2b7f4b1f3b4c31d18107. 
2. I had to manually restore the release's "New Contributors" section automatically generated by GitHub. This script was stripping too much from the automatic release notes. Fixed in #327.

**Pull request merged**